File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/Delphi/Comments/CommentsDemoMain.pas
unit CommentsDemoMain;

{$I cxVer.inc}

interface

uses
  Windows, Messages, SysUtils, Variants, Classes, Graphics, Controls, Forms, Menus, Dialogs, cxGraphics, cxControls,
  cxLookAndFeels, cxLookAndFeelPainters, dxSkinsCore, dxSpreadSheetCore, dxSpreadSheetFunctions, dxSpreadSheetGraphics,
  dxSpreadSheetClasses, dxSpreadSheetTypes, dxSpreadSheet, BaseForm, dxCore, dxCoreClasses, dxHashUtils, ComCtrls,
  dxSpreadSheetCoreHistory, dxSpreadSheetPrinting, dxSpreadSheetFormulas, cxClasses, dxSpreadSheetConditionalFormatting,
  dxSpreadSheetConditionalFormattingRules, dxSpreadSheetContainers, dxSpreadSheetCoreStyles, dxSpreadSheetCoreStrs,
  dxSpreadSheetHyperlinks, dxSpreadSheetStyles, dxSpreadSheetUtils, cxSplitter, dxSpreadSheetFormulaBar;

type
  { TfrmComments }

  TfrmComments = class(TfmBaseForm)
    SpreadSheet: TdxSpreadSheet;
    FormulaBar: TdxSpreadSheetFormulaBar;
    FormulaBarSplitter: TcxSplitter;

    procedure FormCreate(Sender: TObject);
  public
    function GetSpreadSheet: TdxSpreadSheet; override;
  end;

var
  frmComments: TfrmComments;

implementation

{$R *.dfm}

{ TfrmComments }

procedure TfrmComments.FormCreate(Sender: TObject);
begin
  inherited;
  SpreadSheet.LoadFromFile('..\..\Data\Comments_template.xlsx');
  WindowState := wsMaximized;
end;

function TfrmComments.GetSpreadSheet: TdxSpreadSheet;
begin
  Result := SpreadSheet;
end;

end.

File: Components/DevExpressVCL/Demos/ExpressSpreadSheet/CBuilder/Comments/CommentsDemoMain.cpp
//---------------------------------------------------------------------------

#include <vcl.h>
#pragma hdrstop

#include "CommentsDemoMain.h"
//---------------------------------------------------------------------------
#pragma package(smart_init)
#pragma link "BaseForm"
#pragma link "cxClasses"
#pragma link "cxControls"
#pragma link "cxGraphics"
#pragma link "cxLookAndFeelPainters"
#pragma link "cxLookAndFeels"
#pragma link "dxCore"
#pragma link "dxCoreClasses"
#pragma link "dxHashUtils"
#pragma link "dxSpreadSheet"
#pragma link "dxSpreadSheetClasses"
#pragma link "dxSpreadSheetCore"
#pragma link "dxSpreadSheetFormulas"
#pragma link "dxSpreadSheetFunctions"
#pragma link "dxSpreadSheetGraphics"
#pragma link "dxSpreadSheetTypes"
#pragma link "dxSpreadSheetCoreHistory"
#pragma link "dxSpreadSheetPrinting"
#pragma link "dxSpreadSheetConditionalFormatting"
#pragma link "dxSpreadSheetFormulaBar"
#pragma link "cxSplitter"
#pragma resource "*.dfm"
TfrmComments *frmComments;
//---------------------------------------------------------------------------
__fastcall TfrmComments::TfrmComments(TComponent* Owner)
	: TfmBaseForm(Owner)
{
}
//---------------------------------------------------------------------------
void __fastcall TfrmComments::FormCreate(TObject *Sender)
{
  TfmBaseForm::FormCreate(Sender);
  SpreadSheet->LoadFromFile("..\\..\\Data\\Comments_template.xlsx");
  WindowState = wsMaximized;
}
//---------------------------------------------------------------------------

TdxSpreadSheet* TfrmComments::GetSpreadSheet()
{
  return SpreadSheet;
}


